SaraAlert/SaraAlert
# SARAALERT-1260: Allow vaccine table to be populated on import

# Description
Jira Ticket: [SARAALERT-1260](https://tracker.codev.mitre.org/browse/SARAALERT-1260)

- Allow the user to import up to 2 vaccine entries via the Sara Alert format. Similar to how the importing of Lab Results works
- Allow the user to input each of the following data elements per vaccine: Vaccine Group, Product Name, Administration Date & Dose Number.

## Important Changes
Please list important files (meaning substantial or integral to the PR) along with a list of the general changes that should be highlighted for reviewers.

`app/controllers/import_controller.rb`
- Include logic to import vaccines

`app/controllers/patients_controller.rb`
- Allow vaccines to be saved when creating patients

`app/lib/import_export_constants.rb`
- Add new vaccination fields to Sara Alert Format

`public/Sara Alert Import Format.xlsx`
- Update Sara Alert Format guidance

`test/fixtures/files`
- Add new vaccination fields to test fixture files for system tests

`test/system/roles/public_health/dashboard/import_verifier.rb`
- Update system tests to verify vaccination fields

# Checklists

**Submitter:**
- [x] **N/A** If UI changes have been made, Chrome Dev Tools Lighthouse accessibility test has been executed to ensure no 508 issues were introduced.


@Bialogs  :


@ngfreiter 


@dubem7  :
